src/Pages/TemplatesTable/components/AddTemplate/steps/RedhatRepositoriesStep.tsx

```diff
@@ -0,0 +1,324 @@
+        <Text component={TextVariants.h6}>
+          Select additional Red Hat repositories. Core repositories of your OS version have already
+          been added.
+        </Text>
+      </Flex>
+      <Hide hide={(countIsZero && !searchQuery) || isLoading}>
+        <Flex className={classes.topBottomContainers}>
+          <Flex>
+            <FlexItem>
+              <InputGroup>
+                <InputGroupItem isFill>
+                  <TextInput
+                    isDisabled={isLoading}
+                    id='name'
+                    ouiaId='filter_name'
+                    placeholder='Filter by name'
+                    value={searchQuery}
+                    onChange={(_event, value) => setSearchQuery(value)}
+                  />
+                  <InputGroupText isDisabled={isLoading} id='search-icon'>
+                    <SearchIcon />
+                  </InputGroupText>
+                </InputGroupItem>
+              </InputGroup>
+            </FlexItem>
+            <Hide hide={countIsZero}>
+              <FlexItem>
+                <ToggleGroup aria-label='Default with single selectable'>
+                  <ToggleGroupItem
+                    text='All'
+                    buttonId='redhat-repositories-toggle-button'
+                    data-ouia-component-id='all-selected-repositories-toggle'
+                    isSelected={!toggled}
+                    onChange={() => setToggled(false)}
+                  />
+                  <ToggleGroupItem
+                    text='Selected'
+                    buttonId='redhat-repositories-selected-toggle-button'
+                    data-ouia-component-id='redhat-selected-repositories-toggle'
+                    isSelected={toggled}
+                    isDisabled={
+                      selectedRedhatRepos.size - hardcodedRedhatRepositoryUUIDS.size === 0
+                    }
+                    onChange={() => setToggled(true)}
+                  />
+                </ToggleGroup>
+              </FlexItem>
+            </Hide>
+          </Flex>
+          <Hide hide={countIsZero}>
+            <FlexItem>
+              <Pagination
+                id='top-pagination-id'
+                widgetId='topPaginationWidgetId'
+                isDisabled={isLoading}
+                itemCount={count}
+                perPage={perPage}
+                page={page}
+                onSetPage={onSetPage}
+                isCompact
+                onPerPageSelect={onPerPageSelect}
+              />
+            </FlexItem>
+          </Hide>
+        </Flex>
+      </Hide>
+      {showLoader ? (
+        <Bullseye data-ouia-safe={!isFetching} data-ouia-component-id='redhat_repositories_table'>
+          <EmptyTableState
+            notFiltered={searchQuery === ''}
+            clearFilters={() => setSearchQuery('')}
+            itemName='red hat repositories'
+            notFilteredBody='No Red Hat repositories match your desired version and arch'
```

Comment:
```suggestion
            notFilteredBody='No Red Hat repositories match the version and arch'
```
Style Guide does not like "desire" in tech docs.